### PR TITLE
Clean up project build scripts

### PR DESCRIPTION
**Pull request by jrodbx**
- Set compileSdk and targetSdk to 30
- Externalize dependencies and versions into root build.gradle
- Keep AGP and Lint versions in sync
- Clean up plugin ids
- Gradle 6.1.1 => 6.7.1
- AGP 4.0.1 => 27.0.1
- Force Java 8 compatibility
- Kotlin 1.4.10 => 1.4.21
- Merge all repositories blocks into root project

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,20 +1,71 @@
 buildscript {
-  ext.kotlin_version = '1.4.10'
+  ext.versions = [
+    'minSdk': 21,
+    'compileSdk': 30,
+    'targetSdk': 30,
```
Comment (JakeWharton) [inline build.gradle:5]: No real need to define these both. Libraries should only care about compileSdk and for their samples they can just propagate that as the target as well. The library should never propagate a target itself so the values will never deviate.

File: build.gradle [hunk 2]
```diff
@@ -28,3 +79,7 @@ subprojects { project ->
     }
   }
 }
+
+tasks.wrapper {
+  distributionType = Wrapper.DistributionType.ALL
```
Comment (JakeWharton) [inline build.gradle:84]: Why? Bin is best. We don't download the sources for all our dependencies by default, they're downloaded on-demand by the IDE. Keeps CI times quick. Same is true here.

File: build.gradle [hunk 3]
```diff
@@ -1,20 +1,71 @@
...
+    'compileSdk': 30,
+    'targetSdk': 30,
+    'agp': '4.1.1',
+    'lint': '27.1.1',  // = 23.0.0 + agp
+    'kotlin': '1.4.21',
+  ]
+
+  ext.deps = [
+    'kotlin': [
+      'stdlib': [
```
Comment (JakeWharton) [inline build.gradle:13]: Probably can just remove these deps as they're added by default in 1.4.x Kotlin. I'll be surprised if the 7 or 8 deps are actually needed.

File: sample-app/build.gradle [hunk 4]
```diff
@@ -1,23 +1,19 @@
 apply plugin: 'com.android.application'
-apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-android-extensions'
```
Comment (JakeWharton) [inline sample-app/build.gradle:3]: oh_no.png

**Review by JakeWharton — APPROVED** [review]

File: build.gradle (hunk 3 above)
Comment (jrodbx) [reply build.gradle:13]: actually, will need to leave jdk8 here for the lint module, due to this build error: https://pspdfkit.com/blog/2020/how-updating-to-kotlin-14-broke-our-linter-rules/

**Comment by oldergod** [comment]
`AGP 4.0.1 => 27.0.1` ?

**Comment by jrodbx** [comment]
Corrected original description, thank you!

**Pull request merged**